js/navbar.js

```diff
@@ -24,11 +29,25 @@ $(window).scroll(function() {
    cur = cur[cur.length-1];
    var id = cur && cur.length ? cur[0].id : "";
    
-   if (lastId !== id) {
-       lastId = id;
+   if (lastScrollId !== id) {
+       // Set/remove active class
+       menuItems.parent().end().filter("[href='#"+lastScrollId+"']").parent().removeClass("active");
+       lastScrollId = id;
+       menuItems.parent().end().filter("[href='#"+id+"']").parent().addClass("active");
+   }  
+
+  var subCur = subItems.map(function(){
+     if ($(this).offset().top < fromTop)
+       return this;
+   });
+   // Get the id of the current element
+   subCur = subCur[subCur.length-1];
+   var subId = subCur && subCur.length ? subCur[0].id : "";
+   
+   if (lastSubId !== subId) {
        // Set/remove active class
-       menuItems
-         .parent().removeClass("active")
-         .end().filter("[href='#"+id+"']").parent().addClass("active");
-   }                   
+       menuItems.parent().end().filter("[href='#"+lastSubId+"']").removeClass("active");
```

Comment:
I trust you that there is no nicer way to write this :)
I do not 100% understand what it means